node/actors/network/src/gossip/runner.rs

```diff
@@ -113,9 +113,13 @@ impl Network {
             ctx,
             self.cfg.rpc.push_block_store_state_rate,
         );
-        let push_signature_client =
-            rpc::Client::<rpc::push_batch_votes::Rpc>::new(ctx, self.cfg.rpc.push_batch_votes_rate);
-        let push_signature_server = PushBatchVotesServer(self);
+        let mut push_signature_client: Option<rpc::Client<rpc::push_batch_votes::Rpc>> = None;
+        let mut push_signature_server: Option<PushBatchVotesServer> = None;
+        if self.genesis().attesters.is_some() {
+            push_signature_client =
+                rpc::Client::new(ctx, self.cfg.rpc.push_batch_votes_rate).into();
+            push_signature_server = PushBatchVotesServer(self).into();
+        };
```

Comment:
you should be able to create Client/Server adhoc in the if clause below, rather than go through this "Optional" dance.

Reply:
see the if statement with the ping client

Reply:
Yeah, the approach used for the ping client is better. The thing is, I need the client below in [line 190](https://github.com/matter-labs/era-consensus/blob/5f352665542f8db9544578c5b1650317496d696c/node/actors/network/src/gossip/runner.rs#L190) to push the updates to the peer. I could avoid the server creation, but I need a reference to the client at the beginning to effectively use it there.

Reply:
you can put both of these into a single if, no?

Reply:
Right 👍